src/test/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacadeIT.java

```diff
@@ -64,202 +120,3415 @@ public class QuizFacadeIT extends IsaacIntegrationTest {
+        replay(getQuizAssignmentRequest);
+
+        Response getQuizAssignmentResponse = quizFacade.getQuizAssignments(getQuizAssignmentRequest, QUIZ_FACADE_IT_SECONDARY_TEST_GROUP_ID);
+
+        assertEquals(Response.Status.OK.getStatusCode(), getQuizAssignmentResponse.getStatus());
+
+        @SuppressWarnings("unchecked") List<QuizAssignmentDTO> responseBody =
+            (List<QuizAssignmentDTO>) getQuizAssignmentResponse.getEntity();
+        assertEquals(1, responseBody.size());
+      }
+    }
+
+    @Test
+    public void admin() throws JsonProcessingException {
+      Cookie adminSessionCookie = createManualCookieForAdmin();
+      HttpServletRequest getQuizAssignmentRequest = createRequestWithCookies(new Cookie[] {adminSessionCookie});
+      replay(getQuizAssignmentRequest);
+
+      Response getQuizAssignmentResponse = quizFacade.getQuizAssignments(getQuizAssignmentRequest, QUIZ_FACADE_IT_TEST_GROUP_ID);
+
+      assertEquals(Response.Status.OK.getStatusCode(), getQuizAssignmentResponse.getStatus());
+
+      @SuppressWarnings("unchecked") List<QuizAssignmentDTO> responseBody =
+          (List<QuizAssignmentDTO>) getQuizAssignmentResponse.getEntity();
+      assertEquals(6, responseBody.size());
+    }
   }
 
-  @Test
-  public void previewQuizEndpoint_previewHiddenFromRoleTutorQuizAsTutor_fails() throws NoCredentialsAvailableException,
-      NoUserException, SegueDatabaseException, AuthenticationProviderMappingException,
-      IncorrectCredentialsProvidedException, AdditionalAuthenticationRequiredException, InvalidKeySpecException,
-      NoSuchAlgorithmException, MFARequiredButNotConfiguredException {
-    // Arrange
-    // log in as Tutor, create request
-    LoginResult tutorLogin = loginAs(httpSession, TEST_TUTOR_EMAIL, TEST_TUTOR_PASSWORD);
-    HttpServletRequest assignQuizRequest = createRequestWithCookies(new Cookie[] {tutorLogin.cookie});
-    replay(assignQuizRequest);
-
-    // Act
-    // make request
-    Response previewQuizResponse = quizFacade.previewQuiz(createNiceMock(Request.class), assignQuizRequest,
-        QUIZ_HIDDEN_FROM_ROLE_TUTORS_QUIZ_ID);
-
-    // Assert
-    // check status code is FORBIDDEN
-    assertEquals(Response.Status.FORBIDDEN.getStatusCode(), previewQuizResponse.getStatus());
-
-    // check an error message was returned
-    SegueErrorResponse responseBody = (SegueErrorResponse) previewQuizResponse.getEntity();
-    assertEquals("You do not have the permissions to complete this action", responseBody.getErrorMessage());
+  /**
+   * As the integration tests do not currently support MFA login, we cannot use the normal login process and have to
+   * create cookies manually when testing admin accounts.
+   *
+   * @return a Cookie loaded with session information for the test admin user.
+   * @throws JsonProcessingException if the cookie serialisation fails
+   */
+  public Cookie createManualCookieForAdmin() throws JsonProcessingException {
```

Comment:
Shouldn't this be private?

Reply:
I did consider moving this over to the base test class alongside the existing loginAs method for if we need to test other classes with an admin. It's not being used anywhere else _yet_ though, so leaving it in here as private for now could also make sense... Which would you recommend?

Reply:
Unless I know for sure that I'm going to need something in the near future (for instance, because there is a ticket in the current or next sprint that we're definitely going to be working on), I never do things just in case we need them in the future. If you google YAGNI and KISS principles, it will explain it better than I can.

TL;DR: `private`.

Reply:
This function should be moved back to `QuizFacadeIT`, again because of YAGNI and KISS principles. You've moved it to `IsaacIntegrationTest`, but it's only used here, so it's here where it belongs. If at some point you need it in another test, then yes, you can move it to `IsaacIntegrationTest`, or even better, somewhere else and use composition, instead of inheritance, to get that functionality. Does this all make sense? I'm happy to explain further if it doesn't.

Reply:
Understood, I'll move that back and set it to private for now 👍